mqclient/queue.py

```diff
@@ -211,17 +278,34 @@ class MessageGeneratorContext:
         "context has not been entered. Use 'with as' syntax."
     )
 
-    def __init__(self, sub: Sub, queue: Queue) -> None:
+    def __init__(
+        self, sub: Sub, queue: Queue, _span_parent_carrier: Dict[str, Any]
+    ) -> None:
         logging.debug("[MessageGeneratorContext.__init__()]")
         self.sub = sub
         self.message_generator = sub.message_generator(
             timeout=queue.timeout, propagate_error=(not queue.except_errors)
         )
         self.queue = queue
+        self._span_parent_carrier = _span_parent_carrier
+        self._span: Optional[wtt.Span] = None
+
+        self.i = 0
 
         self.entered = False
         self.msg: Optional[Message] = None
 
+    @wtt.spanned(
+        these=[
+            "self.queue._backend_name",
+            "self.queue._address",
+            "self.queue._name",
+            "self.queue._prefetch",
+            "self.queue.timeout",
+        ],
+        carrier="self._span_parent_carrier",
+        behavior=wtt.SpanBehavior.ONLY_END_ON_EXCEPTION,
+    )
```

Comment:
I'm personally unsure if the context manager needs to get spanned, since it doesn't really do much.  It's just error handling for the iterator.

Reply:
I agree a span isn't all that useful on its own. But what it does give us is a common span parent for the iterations.